sdcm/remote/kubernetes_cmd_runner.py

```diff
@@ -530,6 +530,12 @@ def _create_connection(self):
         # NOTE: we increase counter here because each new usage of the "KubernetesPodWatcher" class
         #       instance is triggered by the need to created a new pod with a unique name.
         self._pod_counter += 1
+
+        if self.kluster.params.get('reuse_cluster'):
+            pod_names = self.kluster.kubectl("get pods -o name", namespace=self.namespace).stdout.split()
+            while f'pod/{self.pod_name}' in pod_names:
+                self._pod_counter += 1
```

Comment:
Need to move this fix out of this PR because it is unrelated to the goal of this PR.

Reply:
Well my PR does enable the dynamic loader for the functional test, so it's kind of related.

This specific change doesn't need a PR cycle on it's own (no test code is covering the reuse cases)

Reply:
moved to https://github.com/scylladb/scylla-cluster-tests/pull/5437